source/s3_client.c

```diff
@@ -451,6 +453,8 @@ struct aws_s3_client *aws_s3_client_new(
     return client;
 
 on_error:
+    aws_string_destroy(client->region);
```

Comment:
Good catch, I missed it from the `test_s3_client_create_error`, can you replace [here](https://github.com/awslabs/aws-c-s3/blob/main/tests/s3_data_plane_tests.c#L62) with the following code?
```
    struct aws_s3_client_config client_config = {
        .region = g_test_s3_region,
        .client_bootstrap = tester.client_bootstrap,
    };
```

Reply:
We can't fail the test if we add bootstrap to S3 client.